### PR TITLE
RI-8143: redisearch version fallback

### PR DESCRIPTION
**Pull request by dantovska**
# What

Add a blocking fallback screen on the Vector Search page when the connected database has RediSearch < 2.0 (which ships with Redis < 6). The page relies on `FT._LIST` and other RediSearch 2.0+ commands, so older versions get a clear upgrade prompt instead of errors.
The `hasMinimumRedisearchVersion` check is added to `useRedisInstanceCompatibility` and guards the router before the existing `hasRedisearch` check — so RediSearch 1.x users see the version fallback, while databases with no RediSearch at all continue to see the "RQE not available" screen.
Also refactors `RqeNotAvailable` and the new `VersionNotSupported` into thin wrappers around a shared `SearchPageFallback` component, driven by content constants — eliminating duplicated layout/styling code.

| Light | Dark |
| --- | --- | 
| image  | image |

# Testing

1. Connect to a Redis 5.x instance with RediSearch 1.x — navigate to Search page → should see "Redis Query Engine 2.0+ required" fallback
2. Connect to a Redis 6+ instance without RediSearch → should see "RQE not available" fallback (unchanged behavior)
3. Connect to a Redis 7.2+ instance with RediSearch 2.x → Search page loads normally

---

> [!NOTE]
> **Medium Risk**
> Adds new compatibility gating and version parsing logic that changes when the Vector Search router blocks navigation, which could affect access for some Redis/module combinations. Risk is moderated by expanded unit tests covering RediSearch presence/version edge cases and the new fallback screens.
> 
> **Overview**
> Adds a **blocking “version not supported” fallback** to Vector Search when the connected instance has RediSearch < 2.0 (before the existing “RQE not available” missing-module guard), via a new `hasMinimumRedisearchVersion` flag in `useRedisInstanceCompatibility` and a new `VersionNotSupported` screen.
> 
> Refactors the existing `RqeNotAvailable` UI into a reusable `SearchPageFallback` component driven by content constants (including updated UTM campaign source), and updates/extends tests to cover the new routing behavior and RediSearch version-detection (including integer-to-semver decoding and additional module name aliases).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b580c863f1fe3c1a75478f9b1ac5eaa67aac657e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: redisinsight/ui/src/pages/vector-search/VectorSearchPageRouter.spec.tsx [hunk 1]
```diff
@@ -17,6 +17,14 @@ jest.mock('./components/rqe-not-available', () => {
   }
 })
 
+jest.mock('./components/version-not-supported', () => {
```
Comment (valkirilov) [inline redisinsight/ui/src/pages/vector-search/VectorSearchPageRouter.spec.tsx]: nit: I think we don't have to mock it, the test should pass without it.
